File: Mini Project 0/5e.py
class Node:
    def __init__(self, data = None):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None
    def append(self, node):
        current = self.head
        while (current != None):
            temp = current
            current = current.next
        temp.next = node
    def print_list(self):
        current = self.head
        print("List elements:", end = " ")
        while (current != None):
            print(current.data, end = " ")
            current = current.next
        print("")

llist = LinkedList()
llist.head = Node(1)
second = Node(2)
third = Node(3)
llist.head.next = second
second.next = third

llist.print_list()
fourth = Node(4)
llist.append(fourth)
llist.print_list()

File: Mini Project 0/5e.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node* next;
} node_t;

void print_list(node_t* node) {
    while(node != NULL) {
        printf(" %d ", node->data);
        node = node->next;
    }
}

void free_list(node_t* node) {
    node_t* tmp;
    
    while (node != NULL) {
        tmp = node;
        node = node->next;
        free(tmp);
    }
}

int main(void){
    node_t* head = NULL;
    node_t* second = NULL;
    node_t* third = NULL;
    node_t* fourth = NULL;

    head = (node_t*)malloc(sizeof(node_t));
    second = (node_t*)malloc(sizeof(node_t));
    third = (node_t*)malloc(sizeof(node_t));

    head->data = 1;
    head->next = second;
    second->data = 2;
    second->next = third;
    third->data = 3;
    third->next = NULL;

    printf("Current list elements are:\n");
    print_list(head);

    fourth = (node_t*)malloc(sizeof(node_t));
    third->next = fourth;
    fourth->next = NULL;

    printf("\n\nEnter new integer element to append: ");
    scanf("%d", &(fourth->data));

    printf("\nUpdated list elements are:\n");
    print_list(head);
    printf("\n\n");
    
    free_list(head);
    
    return 0;
}